In [2]:
import pandas as pd

In [3]:
nlp_data1=pd.read_csv('nlp_data1.csv')

### Random Forest Classifier

In [4]:
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
from sklearn.feature_extraction.text import TfidfVectorizer
import joblib

X = nlp_data1['lemmatized_token']
y = nlp_data1['target']

tfidf_vectorizer = TfidfVectorizer()
X = tfidf_vectorizer.fit_transform(X)


X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [5]:
from sklearn.ensemble import RandomForestClassifier

model= RandomForestClassifier()

model.fit(X_train, y_train)

y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred)

print('accuracy', accuracy)
print('precision', precision)
print('recall',recall)
print('f1', f1)

joblib.dump(model, 'random_forest.pkl')
loaded_model = joblib.load('random_forest.pkl')

### Logistic Regression

In [ ]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import confusion_matrix 

model= RandomForestClassifier()

model.fit(X_train, y_train)

y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred)

conf_matrix = confusion_matrix (y_test, y_pred)

print('accuracy', accuracy)
print('precision', precision)
print('recall',recall)
print('f1', f1)
print('cofusion matrix', conf_matrix)

joblib.dump(model, 'random_forest.pkl')
loaded_model = joblib.load('random_forest.pkl')

accuracy 0.7774130006565988
precision 0.7980769230769231
recall 0.6394453004622496
f1 0.7100085543199315
cofusion matrix [[769 105]
 [234 415]]


#### Accuracy:
Accuracy is the proportion of correctly classified instances among all instances.
In this case, the accuracy is approximately 0.7774, or 77.74%. This means that around 77.74% of the instances in the test set were classified correctly by the model.

#### Precision:
Precision is the proportion of true positive predictions among all positive predictions made by the model.
In this case, the precision is approximately 0.7981, or 79.81%. This means that when the model predicts a positive class, it is correct around 79.81% of the time.

#### Recall:
Recall, also known as sensitivity or true positive rate, is the proportion of true positive predictions among all actual positive instances in the data.
In this case, the recall is approximately 0.6394, or 63.94%. This means that the model correctly identifies around 63.94% of all actual positive instances.

#### F1 Score:
The F1 score is the harmonic mean of precision and recall. It provides a balance between precision and recall.
In this case, the F1 score is approximately 0.7100. It takes into account both precision and recall, providing a single score that balances the trade-off between false positives and false negatives.

#### Confusion Matrix:
The confusion matrix provides a more detailed breakdown of the model's predictions.
It is a 2x2 matrix where each cell represents the counts of instances according to their actual and predicted classes.

In this confusion matrix:

The top-left cell (769) represents true negatives (TN), meaning instances that were correctly predicted as negative.
The top-right cell (105) represents false positives (FP), meaning instances that were incorrectly predicted as positive.
The bottom-left cell (234) represents false negatives (FN), meaning instances that were incorrectly predicted as negative.
The bottom-right cell (415) represents true positives (TP), meaning instances that were correctly predicted as positive.